js/core/src/init.ts

```diff
@@ -638,6 +638,7 @@ export function get_component(
 } {
 	let example_component_map: Map<ComponentMeta["type"], LoadingComponent> =
 		new Map();
+	if (type === "api") type = "state";
```

Comment:
instead of setting the type to "state" which could have repercussions for other code, better to have a list of non-rendered components and include both "state" and "api"